include/pybind11/typing.h

```diff
@@ -70,14 +70,31 @@ class Type : public type {
 
 template <typename... Types>
 class Union : public object {
+    PYBIND11_OBJECT_DEFAULT(Union, object, PyObject_Type)
     using object::object;
 };
 
 template <typename T>
 class Optional : public object {
+    PYBIND11_OBJECT_DEFAULT(Optional, object, PyObject_Type)
     using object::object;
 };
 
+#if defined(__cpp_nontype_template_parameter_class)
+template <size_t N>
+struct StringLiteral {
+    constexpr StringLiteral(const char (&str)[N]) { std::copy_n(str, N, name); }
+    char name[N];
+};
+
+// NOTE: C++ implemented constant template Literal[1, 2] does not equal Literal[2, 1]
```

Comment:
Could this be simplified?

```
// NOTE: Literal[1, 2] does not equal Literal[2, 1]
```

But why is this noteworthy here? (IOW, where/why does it matter?)

Reply:
Oops this comment was left from a previous implementation. It has now been removed.